AdobeStockClient/Model/ConnectionAdapter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockClient\Model;
+
+use AdobeStock\Api\Client\AdobeStock;
+use Magento\AdobeStockClientApi\Api\ConnectionAdapterInterface;
+
+/**
+ * Class ConnectionAdapter is responsible for connection initialization to the Adobe Stock service.
+ */
+class ConnectionAdapter implements ConnectionAdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var string
+     */
+    private $productName;
+
+    /**
+     * @var string
+     */
+    private $targetEnvironment;
+
+    /**
+     * @inheritdoc
+     */
+    public function initializeConnection(): AdobeStock
+    {
+        $connection = new AdobeStock(
```

Comment:
It's not a good practice to introduce temporal coupling. Your initialization method depends on data that should be provided before the method call.